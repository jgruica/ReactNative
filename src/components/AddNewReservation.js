import React, { Component } from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  Picker
} from 'react-native';
import moment from 'moment';
import DateTimePicker from 'react-native-modal-datetime-picker';
import Loader from './Loader';
import { graphql, compose, withApollo } from 'react-apollo';
import { createReservation } from '../../mutations/AllMutations';
import { reservations } from '../../queries/AllQueries';

import Icon from 'react-native-vector-icons/Entypo';

const styles = {
  wrapper: {
    flex: 1,
    paddingTop: 23,
    alignItems: 'stretch'
  },
  text: {
    fontSize: 15,
    textAlign: 'left',
    marginLeft: 10,
    color: '#2e3131'
  },
  input: {
    margin: 10,
    height: 40,
    borderColor: '#8c14fc',
    borderWidth: 1
  },
  submitButton: {
    backgroundColor: '#8c14fc',
    padding: 10,
    margin: 15,
    height: 40,
  },
  submitButtonText: {
    color: 'white'
  }
}

class AddNewReservation extends Component {
  state = {
    name: '',
    hotelName: '',
    arrivalDate: '',
    departureDate: '',
    creatingReservation: false,
    creatingReservationErr: "",
    arrivalDatePickerVisible: false,
    departureDatePickerVisible: false,
  }

  createNewReservation = () => {
    const { name, hotelName, arrivalDate, departureDate } = this.state;
    if (!name.length || !hotelName.length || !arrivalDate || !departureDate) {
      this.setState({ creatingReservationErr: "Fields can't be empty", creatingReservation: false });
      return;
    }
    this.setState({ creatingReservation: true });
    this.props.createReservation({ name, hotelName, arrivalDate, departureDate })
      .then(() => {
        this.setState({ creatingReservation: false });
        this.props.history.push('/');
      })
      .catch(() => {
        this.setState({ creatingReservation: false });
      });
  }

  showDateTimePicker = (field) => {
    this.setState({ [field]: true });
  }

  hideDateTimePicker = (field) => {
    this.setState({ [field]: false });
  }

  handleDatePicked = (date, dateField, hideField) => {
    this.setState({
      [dateField]: date
    })
    this.hideDateTimePicker(hideField);
  };


  render() {
    const { creatingReservationErr, creatingReservation } = this.state;
    const { data } = this.props;
    
    const arrivalDateValue = this.state.arrivalDate !== '' ?
      moment(this.state.arrivalDate).format('l').toString() : '';
    const departureDateValue = this.state.departureDate !== '' ?
      moment(this.state.departureDate).format('l').toString() : '';


    return (
      <View style={styles.wrapper}>
        {creatingReservationErr ? <Text>{creatingReservationErr}</Text> : null}
        <Text style={styles.text}>Name:</Text>
        <TextInput
          style={styles.input}
          placeholder="Name"
          onChangeText={(name) => this.setState({ name })} />
        <Text style={styles.text}>Pick hotel: </Text>
        {data.loading ? 
         <Loader loading={data.loading} />
        :
        <Picker
        selectedValue={this.state.hotelName}
        onValueChange={(itemValue, itemIndex) =>
          this.setState({ hotelName: itemValue })
        }>
        {data.reservations.map(reservation => {
          return (
            <Picker.Item label={reservation.hotelName} value={reservation.hotelName} />
          )
        })}
      </Picker>
        }
        <TouchableOpacity onPress={() => this.showDateTimePicker('arrivalDatePickerVisible')}>
          <TextInput
            style={styles.input}
            placeholder="Choose arrival time"
            value={arrivalDateValue}
            editable={false}
            pointerEvents="none"
          />
        </TouchableOpacity>
        <DateTimePicker
          isVisible={this.state.arrivalDatePickerVisible}
          mode='date'
          onConfirm={(date) => this.handleDatePicked(date, 'arrivalDate', 'arrivalDatePickerVisible')}
          onCancel={() => this.hideDateTimePicker('arrivalDatePickerVisible')}
        />
        <TouchableOpacity onPress={() => this.showDateTimePicker('departureDatePickerVisible')}>
          <TextInput
            style={styles.input}
            placeholder="Choose departure time"
            value={departureDateValue}
            editable={false}
            pointerEvents="none"
          />
        </TouchableOpacity>
        <DateTimePicker
          isVisible={this.state.departureDatePickerVisible}
          mode='date'
          onConfirm={(date) => this.handleDatePicked(date, 'departureDate', 'departureDatePickerVisible')}
          onCancel={() => this.hideDateTimePicker('departureDatePickerVisible')}
        />

        <TouchableOpacity
          style={styles.submitButton}
          onPress={this.createNewReservation}
        >
          <Text style={styles.submitButtonText}>{!creatingReservation ? 'Add reservation' : 'Adding...'} </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default compose(
  withApollo,
  graphql(createReservation, {
    props: ({ mutate }) => ({
      createReservation: (data) => mutate({ variables: { data } })
    })
  }),
  graphql(reservations, {
    options: {
      fetchPolicy: "network-only"
    }
  }),
)(AddNewReservation);