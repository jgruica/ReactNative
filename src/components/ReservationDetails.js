import React, { Component } from 'react';
import {
  Text,
  TextInput,
  View,
  TouchableOpacity,
  Modal,
  TouchableHighlight
} from 'react-native';
import { graphql, compose, withApollo } from 'react-apollo';

import moment from 'moment';
import DateTimePicker from 'react-native-modal-datetime-picker';

import Loader from "./Loader";
import { reservationQuery } from '../../queries/AllQueries';
import { updateReservation, deleteReservation } from '../../mutations/AllMutations';

import styles from "../../styles/ReservationDetailsStyle";

export class ReservationDetails extends Component {
  state = {
    name: '',
    hotelName: '',
    arrivalDate: '',
    departureDate: '',
    displayDeleteModal: false,
    updatingReservation: false,
    updatingReservationErr: '',
    arrivalDatePickerVisible: false,
    departureDatePickerVisible: false,
  }

  static navigationOptions = {
    title: "Details",
  }

  toggleDeleteModal = () => {
    this.setState({ displayDeleteModal: !this.state.displayDeleteModal })
  }

  onPressDelete = () => {
    this.toggleDeleteModal();
  }

  componentDidMount() {
    if (this.props.data.reservation) {
      this.setState({
        name: this.props.data.reservation.name,
        hotelName: this.props.data.reservation.hotelName,
        arrivalDate: this.props.data.reservation.arrivalDate,
        departureDate: this.props.data.reservation.departureDate
      })
    }
  }

  componentDidUpdate(lastProps) {
    if (this.props.data.reservation !== lastProps.data.reservation && this.props.data.reservation) {
      this.setState({
        name: this.props.data.reservation.name,
        hotelName: this.props.data.reservation.hotelName,
        arrivalDate: this.props.data.reservation.arrivalDate,
        departureDate: this.props.data.reservation.departureDate
      })
    }
  }

  updateReservation = () => {
    const { name, hotelName, arrivalDate, departureDate } = this.state;
    if (!name.length || !hotelName.length || !arrivalDate.length || !departureDate.length) {
      this.setState({ updatingReservationErr: "Fields can't be empty", updatingReservation: false });
      return;
    }
    this.setState({ updatingReservation: true });
    this.props.updateReservation({ name, hotelName, arrivalDate, departureDate }, { id: this.props.navigation.getParam('id') })
      .then(() => {
        this.setState({ updatingReservation: false });
        this.props.navigation.navigate('Home');
      })
      .catch(() => {
        this.setState({ updatingReservation: false });
      });
  }

  deleteReservation = () => {
    this.props.deleteReservation({ id: this.props.navigation.getParam('id') })
      .then(() => {
        this.toggleDeleteModal();
        this.props.navigation.navigate('Home');
      })
      .catch();
  }

  showDateTimePicker = (field) => {
    this.setState({ [field]: true });
  }

  hideDateTimePicker = (field) => {
    this.setState({ [field]: false });
  }

  handleDatePicked = (date, dateField, hideField) => {
    this.setState({
      [dateField]: date
    })
    this.hideDateTimePicker(hideField);
  };

  render() {
    const { updatingReservationErr } = this.state;
    const { data } = this.props;

    const arrivalDateValue = this.state.arrivalDate !== '' ?
      moment(this.state.arrivalDate).format('l').toString() : '';
    const departureDateValue = this.state.departureDate !== '' ?
      moment(this.state.departureDate).format('l').toString() : '';

    return (
      <View style={styles.wrapper}>
        {data.loading ?
          <Loader loading={data.loading} />
          :
          <View style={{ padding: 5 }}>
            {updatingReservationErr ? <Text>{updatingReservationErr}</Text> : null}
            <Text style={styles.text}>Name:</Text>
            <TextInput
              style={styles.input}
              defaultValue={data.reservation.name}
              onChangeText={(name) => this.setState({ name })} />
            <Text style={styles.text}>Hotel name:</Text>
            <TextInput
              style={styles.input}
              defaultValue={data.reservation.hotelName}
              onChangeText={(hotelName) => this.setState({ hotelName })} />
            <Text style={styles.text}>Arrival time:</Text>
            <TouchableOpacity onPress={() => this.showDateTimePicker('arrivalDatePickerVisible')}>
              <TextInput
                style={styles.input}
                defaultValue={arrivalDateValue}
                editable={false}
                pointerEvents="none"
              />
            </TouchableOpacity>
            <DateTimePicker
              isVisible={this.state.arrivalDatePickerVisible}
              mode='date'
              onConfirm={(date) => this.handleDatePicked(date, 'arrivalDate', 'arrivalDatePickerVisible')}
              onCancel={() => this.hideDateTimePicker('arrivalDatePickerVisible')}
            />
              <Text style={styles.text}>Departure time:</Text>
            <TouchableOpacity onPress={() => this.showDateTimePicker('departureDatePickerVisible')}>
              <TextInput
                style={styles.input}
                defaultValue={departureDateValue}
                editable={false}
                pointerEvents="none"
              />
            </TouchableOpacity>
            <DateTimePicker
              isVisible={this.state.departureDatePickerVisible}
              mode='date'
              onConfirm={(date) => this.handleDatePicked(date, 'departureDate', 'departureDatePickerVisible')}
              onCancel={() => this.hideDateTimePicker('departureDatePickerVisible')}
            />
          </View>
        }
        <View style={styles.buttonsContainer}>
        <TouchableOpacity
            style={styles.deleteButton}
            onPress={this.onPressDelete}
          >
            <Text style={styles.buttonText}>Delete</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.updateButton}
            onPress={this.updateReservation}
          >
            <Text style={styles.updateButtonText}>Update</Text>
          </TouchableOpacity>
        </View>
        {this.state.displayDeleteModal &&
          <Modal
            animationType="fade"
            transparent={true}
            visible={this.state.displayDeleteModal}
            onRequestClose={() => {
              Alert.alert('Modal has been closed.');
            }}>
            <View style={styles.modalContent}>
              <View>
                <Text style={styles.text}>Are you sure you want to delete this reservation? </Text>
                <View style={styles.buttonsContainer}>
                  <TouchableHighlight
                    style={styles.cancelButon}
                    onPress={() => {
                      this.toggleDeleteModal();
                    }}>
                    <Text style={styles.cancelText}>Cancel</Text>
                  </TouchableHighlight>
                  <TouchableHighlight
                    style={styles.deleteButton}
                    onPress={this.deleteReservation}>
                    <Text style={styles.buttonText}>Delete</Text>
                  </TouchableHighlight>
                </View>
              </View>
            </View>
          </Modal>
        }
      </View>

    )
  }
}


export default compose(
  withApollo,
  graphql(reservationQuery, {
    options: ({ navigation }) => ({
      variables: { where: { id: navigation.getParam('id')} }
    })
  }),
  graphql(updateReservation, {
    props: ({ mutate }) => ({
      updateReservation: (data, where) => mutate({ variables: { data, where } })
    })
  }),
  graphql(deleteReservation, {
    props: ({ mutate }) => ({
      deleteReservation: (where) => mutate({ variables: { where } })
    })
  }),
)(ReservationDetails);