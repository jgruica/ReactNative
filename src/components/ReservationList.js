import React, { Component } from 'react';
import {
  Text,
  View,
  ScrollView,
  TouchableOpacity
} from 'react-native';
import { graphql, compose, withApollo } from 'react-apollo';
import Icon from 'react-native-vector-icons/Entypo';

import moment from 'moment';
import sortBy from 'lodash/sortBy';

import Loader from './Loader';

import { reservations } from '../../queries/AllQueries';

import styles from "../../styles/ReservationListStyle";

const colors = ['#f1a9a0', '#d2527f', '#e74c3c', '#9f5afd', 
'#913d88', '#81cfe0', '#1e8bc3', '#2ecc71', '#1e824c', '#fef160', '#e67e22', '#f5e51b', '#f27935']
let nextColorIndex = 0
const colorMap = {}
const getHotelColor = (hotelName) => {
  if (!colorMap[hotelName]) {
    colorMap[hotelName] = colors[nextColorIndex]
    nextColorIndex = (nextColorIndex + 1) % colors.length
  }
  return {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    borderBottomColor: '#A9A9A9',
    borderBottomWidth: 0.2,
    borderLeftColor: colorMap[hotelName],
    borderLeftWidth: 10,
  }
}

class ReservationList extends Component {
  static navigationOptions = {
    title: "Reservations",
  }
  render() {
    const { data } = this.props;
    const sortedReservations = data.reservations && data.reservations.length && sortBy(data.reservations, "hotelName");
    const detailsIcon = <Icon name="chevron-small-right" size={30} color="#A9A9A9" />;
    if (data.loading) {
      return <Loader loading={data.loading} />;
    } else {
      return (
        <View>
          <ScrollView>
            {sortedReservations.map(reservation =>
              <View key={reservation.id} style={getHotelColor(reservation.hotelName)} >
                <Text style={styles.header}>{reservation.name}</Text>
                <Text style={styles.text}>{moment(reservation.arrivalDate).format("ll")} - {moment(reservation.departureDate).format("ll")}</Text>
                <Text style={styles.text}>{reservation.hotelName}</Text>
                <TouchableOpacity onPress={() =>
                  this.props.navigation.navigate('Details', { id: reservation.id })}
                  style={{ position: 'absolute', top: 10, right: 20 }}>
                  <Text>{detailsIcon}</Text>
                </TouchableOpacity>
              </View>
            )}
          </ScrollView>
          <TouchableOpacity onPress={() =>
            this.props.navigation.navigate('AddNew')
          } style={{
            width: 60, height: 60, borderRadius: 60 / 2, backgroundColor: '#8c14fc',
            position: 'absolute', bottom: 30, right: 20, alignSelf: 'flex-end'
          }} underlayColor="#f0f4f7">
            <Text style={{ fontSize: 25, color: '#FFFFFF', left: 22, top: 12 }}>+</Text>
          </TouchableOpacity>
        </View>
      );
    }
  }
}

export default compose(
  withApollo,
  graphql(reservations, {
    options: {
      fetchPolicy: "network-only"
    }
  }),
)(ReservationList);