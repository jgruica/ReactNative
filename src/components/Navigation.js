import { createStackNavigator, createAppContainer } from 'react-navigation';
import ReservationList from './components/ReservationList';
import ReservationDetails from './components/ReservationDetails';
import AddNewReservation from './components/AddNewReservation';

const Root = createStackNavigator({
    Home: { 
      screen: ReservationList,
    },
    Details: { 
      screen: ReservationDetails,
    },
    AddNew: {
      screen: AddNewReservation
    }
});

const Navigation = createAppContainer(Root);
export default Navigation; 