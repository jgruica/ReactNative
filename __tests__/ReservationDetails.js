import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import { ReservationDetails } from '../src/components/ReservationDetails';
jest.mock('DatePickerIOS', () => 'DatePickerIOS');

describe('Reservation details component', () => {
  const data = {
    reservation: [{
      id: '123',
      name: 'Test Name',
      hotelName: 'Test Hotel',
      arrivalDate: '01/01/2020',
      departureDate: '01/05/2020'
    }]
  }

  it('Details: renders correctly', () => {
    const tree = renderer.create(<ReservationDetails data={data}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});