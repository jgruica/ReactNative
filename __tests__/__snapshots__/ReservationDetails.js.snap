// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation details component Details: renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2e3131",
          "fontSize": 15,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      Name:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#8c14fc",
          "borderRadius": 10,
          "borderWidth": 0.2,
          "height": 40,
          "margin": 10,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#2e3131",
          "fontSize": 15,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      Hotel name:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "#8c14fc",
          "borderRadius": 10,
          "borderWidth": 0.2,
          "height": 40,
          "margin": 10,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#2e3131",
          "fontSize": 15,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      Arrival time:
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="3/14/2019"
        editable={false}
        pointerEvents="none"
        style={
          Object {
            "borderColor": "#8c14fc",
            "borderRadius": 10,
            "borderWidth": 0.2,
            "height": 40,
            "margin": 10,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <DatePickerIOS
              date={2019-03-15T03:31:22.460Z}
              minuteInterval={1}
              mode="date"
              onCancel={[Function]}
              onConfirm={[Function]}
              onDateChange={[Function]}
              onHideAfterConfirm={[Function]}
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#d5d5d5",
                "borderTopWidth": 0.5,
                "height": 57,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "height": 57,
              "justifyContent": "center",
              "marginBottom": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
    <Text
      style={
        Object {
          "color": "#2e3131",
          "fontSize": 15,
          "marginLeft": 10,
          "textAlign": "left",
        }
      }
    >
      Departure time:
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="3/14/2019"
        editable={false}
        pointerEvents="none"
        style={
          Object {
            "borderColor": "#8c14fc",
            "borderRadius": 10,
            "borderWidth": 0.2,
            "height": 40,
            "margin": 10,
            "paddingLeft": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        pointerEvents="box-none"
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <DatePickerIOS
              date={2019-03-15T03:31:22.460Z}
              minuteInterval={1}
              mode="date"
              onCancel={[Function]}
              onConfirm={[Function]}
              onDateChange={[Function]}
              onHideAfterConfirm={[Function]}
            />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#d5d5d5",
                "borderTopWidth": 0.5,
                "height": 57,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "height": 57,
              "justifyContent": "center",
              "marginBottom": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 20,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#cf000f",
          "borderRadius": 10,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Delete
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#8c14fc",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#8c14fc",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Update
      </Text>
    </View>
  </View>
</View>
`;
